rcl/test/rcl/test_node.cpp

```diff
@@ -511,15 +511,21 @@ TEST_F(CLASSNAME(TestNodeFixture, RMW_IMPLEMENTATION), test_rcl_node_init_with_i
     EXPECT_EQ(RCL_RET_ERROR, ret);
     rcl_reset_error();
   }
+
   // Try normal init but force an internal error on fini.
   {
     ret = rcl_node_init(&node, name, namespace_, &context, &options);
     EXPECT_EQ(RCL_RET_OK, ret);
-    auto mock = mocking_utils::inject_on_return("lib:rcl", rmw_destroy_node, RMW_RET_ERROR);
-    ret = rcl_node_fini(&node);
-    EXPECT_EQ(RCL_RET_ERROR, ret);
+    auto mock = mocking_utils::patch_and_return(
```

Comment:
Does this match the coverage? Seems like rcl_node_fini would exit elsewhere with this new patch?

Reply:
That's a good question.

I suspect that the coverage will remain the same; we weren't previously covering the failure path when `rcl_logging_rosout_fini_publisher_for_node` failed, so we should be swapping one error path for another.  But I'll run a local coverage build to check for sure.

Reply:
I ran coverage before and after this change, and it is as I thought; we are trading coverage on one line for another.  In both cases, just running the unit tests, we are covering 221/233 lines in src/rcl/node.c